cloud-sql-proxy/cloud-sql-proxy.sh

```diff
@@ -458,36 +464,57 @@ EOF
-  fi
+  log 'Initialzing MySQL DB for Hive metastore ...'
+  local admin_defaults_file="/dev/shm/admin-db.cnf"
+  local hive_defaults_file="/dev/shm/hive-db.cnf"
+  local db_password_param="--defaults-file=${admin_defaults_file}"
+  local db_hive_password_param="--defaults-file=${hive_defaults_file}"
+
+  (
+    printf "[client]\nhost=127.0.0.1\nport=${METASTORE_PROXY_PORT}\nuser=${DB_ADMIN_USER}\npassword=\""
+    perl -pe 'chomp' < /dev/shm/db-pw
+    echo '"'
+  ) > "${admin_defaults_file}"
+  (
+    printf "[client]\nhost=127.0.0.1\nport=${METASTORE_PROXY_PORT}\nuser=${DB_HIVE_USER}\npassword=\""
+    perl -pe 'chomp' < /dev/shm/hive-pw
+    echo '"'
+  ) > "${hive_defaults_file}"
+  (
+    echo -n "CREATE USER IF NOT EXISTS '${DB_HIVE_USER}'@'cloudsqlproxy~%' IDENTIFIED BY '"
+    perl -pe 'chomp' < /dev/shm/hive-pw
+    echo -n "';"
+  ) > /dev/shm/create_hive_user.sql
+
+  # create hive user if it does not exist
+  mysql "${db_hive_password_param}" -e '' || \
+    mysql "${db_password_param}" < /dev/shm/create_hive_user.sql
 
   # Check if metastore is initialized.
-  if ! mysql -h 127.0.0.1 -P "${METASTORE_PROXY_PORT}" -u "${DB_HIVE_USER}" "${db_hive_password_param}" -e ''; then
-    mysql -h 127.0.0.1 -P "${METASTORE_PROXY_PORT}" -u "${DB_ADMIN_USER}" "${db_password_param}" -e \
-      "CREATE USER '${DB_HIVE_USER}' IDENTIFIED BY '${DB_HIVE_PASSWORD}';"
-  fi
-  if ! mysql -h 127.0.0.1 -P "${METASTORE_PROXY_PORT}" -u "${DB_HIVE_USER}" "${db_hive_password_param}" -e "use ${METASTORE_DB}"; then
+  if ! mysql "${db_hive_password_param}" -e "use ${METASTORE_DB}"; then
     # Initialize a Hive metastore DB
-    mysql -h 127.0.0.1 -P "${METASTORE_PROXY_PORT}" -u "${DB_ADMIN_USER}" "${db_password_param}" -e \
-      "CREATE DATABASE ${METASTORE_DB};
-       GRANT ALL PRIVILEGES ON ${METASTORE_DB}.* TO '${DB_HIVE_USER}';"
-    /usr/lib/hive/bin/schematool -dbType mysql -initSchema ||
+    mysql "${db_password_param}" -e \
+      "CREATE DATABASE IF NOT EXISTS ${METASTORE_DB};
+       GRANT ALL PRIVILEGES ON ${METASTORE_DB}.* TO '${DB_HIVE_USER}'@'cloudsqlproxy~%';"
+    /usr/lib/hive/bin/schematool -dbType mysql -initSchema || {
+      rm -f /dev/shm/*-db.cnf /dev/shm/*.sql
       err 'Failed to set mysql schema.'
+    }
   fi
   log 'MYSQL DB initialized for Hive metastore'
+  rm -f /dev/shm/*-db.cnf /dev/shm/*.sql
 }
 
+function exit_handler() {
+  rm -f /dev/shm/*-pw /dev/shm/*-db.cnf /dev/shm/*.sql
+}
+
+trap exit_handler EXIT
+
 function initialize_postgres_metastore_db() {
   log 'Initialzing POSTGRES DB for Hive metastore ...'
-  local admin_connection=postgresql://"${DB_ADMIN_USER}":"${DB_ADMIN_PASSWORD}"@127.0.0.1:"${METASTORE_PROXY_PORT}"/
-  local hive_connection=postgresql://"${DB_HIVE_USER}":"${DB_HIVE_PASSWORD}"@127.0.0.1:"${METASTORE_PROXY_PORT}"/postgres
+  local admin_connection=postgresql://"${DB_ADMIN_USER}":"$(perl -pe 'chomp' < /dev/shm/db-pw)"@127.0.0.1:"${METASTORE_PROXY_PORT}"/
```

Comment:
I was mistaken, after the `wc` change above, these two lines will be the final instance of assignment of classified material to bash environment variable.  I'm not certain how best to address this problem, but I will stand up a PG instance to see if there's a way to specify DSN components from a config file as is done for mysql with `--defaults-file=${admin_defaults_file}`